c/authService.c

```diff
@@ -173,6 +174,76 @@ static int serveAuthCheck(HttpService *service, HttpResponse *res) {
   return 0;
 }
 
+int resetPassword(HttpService *service, HttpResponse *response) {
+  int returnCode = 0, reasonCode = 0;
+  jsonPrinter *p = respondWithJsonPrinter(response);
+  HttpRequest *request = response->request;
+  jsonStart(p);
+
+  if (!strcmp(request->method, methodPOST)) {
+    char *inPtr = request->contentBody;
+    char *nativeBody = copyStringToNative(request->slh, inPtr, strlen(inPtr));
+    int inLen = nativeBody == NULL ? 0 : strlen(nativeBody);
+    char errBuf[JSON_ERROR_BUFFER_SIZE];
+
+    Json *body = jsonParseUnterminatedString(request->slh, nativeBody, inLen, errBuf, JSON_ERROR_BUFFER_SIZE);
+    JsonObject *inputMessage = jsonAsObject(body);
+    Json *username = jsonObjectGetPropertyValue(inputMessage,"username");
+    Json *password = jsonObjectGetPropertyValue(inputMessage,"password");
+    Json *newPassword = jsonObjectGetPropertyValue(inputMessage,"newPassword");
+    int newPasswordLength = strlen(jsonAsString(newPassword));
+
+    if (newPasswordLength == 0) {
+      setResponseStatus(response, HTTP_STATUS_BAD_REQUEST, "ERROR");
+      jsonAddString(p, "status", "No new password provided");
+    } else {
+      resetZosUserPassword(jsonAsString(username),  jsonAsString(password), jsonAsString(newPassword), &returnCode, &reasonCode);
+
+      switch (returnCode) {
+        case PASSWORD_RESET_OK:
+          setResponseStatus(response, HTTP_STATUS_OK, "OK");
+          jsonAddString(p, "status", "Password Successfully Reset");
```

Comment:
I don't think you can start writing using a json printer before writing the headers.